### ISSUE TITLE
Chinese font rendering is blurring.

### ISSUE THREAD

**Issue by AllenDang** [Immediate-Mode-UI/Nuklear#28]
![截屏2019-12-19下午5 13 16](https://user-images.githubusercontent.com/18550/71160610-dbfb6500-2282-11ea-8a76-1250eba565fc.png)

English font rendering is pretty good. But Chinese character rendering is blurring.

I tried to set the oversimple other than 1, I got black squares.

![截屏2019-12-19下午5 15 39](https://user-images.githubusercontent.com/18550/71160820-31d00d00-2283-11ea-9f2b-591b0fe0f72f.png)

**Comment by dumblob** [comment]
As mentioned in Credits in Readme.md, by default Nuklear embeds [stb_truetype](https://github.com/nothings/stb/blob/master/stb_truetype.h ) for basic font rendering, but you're free to use e.g. freetype (there might be some examples in this repo IIRC).

On the other hand, it seems `stb_truetype` [wasn't updated for quite long](https://github.com/Immediate-Mode-UI/Nuklear/blob/bacdca77720e9144b05393c722a0d2b62ced7fed/src/nuklear_font.c#L379 ) in Nuklear and upstream has a new version where chances are, that the blurring will be less significant.

**Comment by AllenDang** [comment]
@dumblob Thanks for your reply. I checked those two codes, is there any utils to generate from stb_truetype? So far as I see, the code is hand write line by line, not just embed directly. It seems to update is not an easy work. Am I correct?

**Comment by dumblob** [comment]
@AllenDang yeah, you're right - feel free to make a PR and separate stb_truetype completely (we use the Python packing script in the same folder through `paq.sh`/`paq.bat` - see also [minor changes to this "amalgamation"](https://github.com/Immediate-Mode-UI/Nuklear/pull/13/files ) which we didn't have time yet to finally review and merge).

**Comment by kiwidoggie** [comment]
@AllenDang how are you rendering the Chinese font's. I'm having an issue from wchar_t to char conversion using the nk_label's.

**Comment by AllenDang** [comment]
@kiwidoggie I'm using Golang binding so I guess I cannot help you one this...

**Comment by AllenDang** [comment]
@dumblob I'm trying to update stb_truetype embeded in nuklear_font.h and I find out that I should do a lot of function and struct naming replacement there.
I'm wondering how to you do it at first place? All hand-writing or there is a script which does the "search-and-replace" work?

**Comment by SairesArt** [comment]
I have had issues with blurry fonts as well.
What fixed it for me is finding the correct scaling value. For whatever reason, in my testing each font has a scaling value, where the font is sharp. These sharp scaling values come in steps of integers.
One font always remains sharp in steps of 4, size 24, 28, 32 etc... whilst one font only remains sharp in steps of 6, size 19, 25, 31 etc...  All other values make stb_truetype produce blurry results.
All of this with oversample_v = 1, oversample_h = 1 and pixel_snap = 1.
Cranking up oversampling only helps a teeny tiny bit whilst costing a bunch of memory.

Hope this will be fixed with the new exellent PR https://github.com/Immediate-Mode-UI/Nuklear/pull/53

**Comment by dumblob** [comment]
@AllenDang could you retest if the update of stb_truetype (see PR https://github.com/Immediate-Mode-UI/Nuklear/pull/53 ) solved your problem?
